packages/node-cli/src/commands/DeployCommandGenerator.js

```diff
@@ -136,60 +136,64 @@ module.exports = class DeployCommandGenerator extends BaseCommandGenerator {
 	}
 
 	async _executeAction(answers) {
-		const SDKParams = CommandUtils.extractCommandOptions(answers, this._commandMetadata);
-		const flags = [COMMAND.FLAGS.NO_PREVIEW, COMMAND.FLAGS.SKIP_WARNING];
-		if (SDKParams[COMMAND.FLAGS.VALIDATE]) {
-			delete SDKParams[COMMAND.FLAGS.VALIDATE];
-			flags.push(COMMAND.FLAGS.VALIDATE);
+		try {
+			const SDKParams = CommandUtils.extractCommandOptions(answers, this._commandMetadata);
+			const flags = [COMMAND.FLAGS.NO_PREVIEW, COMMAND.FLAGS.SKIP_WARNING];
+			if (SDKParams[COMMAND.FLAGS.VALIDATE]) {
+				delete SDKParams[COMMAND.FLAGS.VALIDATE];
+				flags.push(COMMAND.FLAGS.VALIDATE);
+			}
+			const executionContextForDeploy = new SDKExecutionContext({
+				command: this._commandMetadata.sdkCommand,
+				includeProjectDefaultAuthId: true,
+				params: SDKParams,
+				flags,
+			});
+
+			const operationResult = await executeWithSpinner({
+				action: this._sdkExecutor.execute(executionContextForDeploy),
+				message: TranslationService.getMessage(MESSAGES.DEPLOYING),
+			});
+
+			var isValidate = SDKParams[COMMAND.FLAGS.VALIDATE] ? true : false;
+			var isApplyContentProtection =
+				(this._projectType === PROJECT_SUITEAPP && SDKParams[COMMAND.OPTIONS.APPLY_CONTENT_PROTECTION] === SDK_TRUE)
+					? true
+					: false;
+
+			return DeployActionResultMapper.createActionResultFrom(operationResult, isValidate, isApplyContentProtection);
```

Comment:
I think we should have one mapper that is able to create an ActionResult, passing the action name and the operationResult plus some extra parameters. Something, like ActionResultMapper('deploy',operationResult, params). Or ActionResultMapper('deploy')(operationResult, params).

Reply:
hmm ok... it could be like a Factory.... I'll try it.